### PR TITLE
Added tests for assignment actions + redux hookup of evaluation screen

### PR DESCRIPTION
**Pull request by ElyDev**
Initial prep of evaluation page by hooking it up to Redux.
Added tests for new redux actions related to assignments (add, update, and complete assignment).

### CODE REVIEW

**Comment by ElyDev** [comment]
I also implemented initial UI for the Feedback Page.. (See below)
Screenshot_1553682533

**Comment by ElyDev** [comment]
Unfortunately none of the components provided by the react elements' Rating or AirBnbRating controls satis fully fully our needs. They are both missing capabilities that we'd need. For now I've used them to enable the MVP, but we eventually need to use something different of extend them to meet our full design.

**Comment by ElyDev** [comment]
Added snapshot test for the Evaluation page

File: Dev/QConnect/src/model/reducers/classReducer.js [hunk 1]
```diff
@@ -276,8 +276,13 @@ export const classReducer = (state = INITIAL_STATE, action) => {
     case actionTypes.EDIT_CURRENT_ASSIGNMENT:
       classIndex = action.classIndex;
       studentIndex = action.studentIndex;
-      newAssignment = action.newAssignment;
-      newState = update(baseState, { teachers: { [0] : { classes: { [classIndex]: { students: { [studentIndex]: { currentAssignment: { name: { $set: newAssignment } } } } } } } } });
+
```
Comment (zyadelgohary) [inline Dev/QConnect/src/model/reducers/classReducer.js]: Good change, more organized. We should also revise our other code for the reducer actions and simplify them... the attendance action in particular.

File: Dev/QConnect/src/model/reducers/classReducer.js [hunk 2]
```diff
@@ -306,7 +311,7 @@ export const classReducer = (state = INITIAL_STATE, action) => {
       }
       //pushes the assignment to the array of assignment history (Remember, this action does not 
       //update the current assignment, this needs to be done using the addNewAssignment action)
-      newState = update(baseState, { teachers: { [0] : { classes: { [classIndex]: { students: { [studentIndex]: { assignmentHistory: { $push: assignment } } } } } } } });
+      newState = update(baseState, { teachers: { [0] : { classes: { [classIndex]: { students: { [studentIndex]: { assignmentHistory: { $push: [assignment] } } } } } } } });
```
Comment (zyadelgohary) [inline Dev/QConnect/src/model/reducers/classReducer.js]: Good catch, my bad

File: Dev/QConnect/src/screens/Evaluation/EvaluationPage.js [hunk 3]
```diff
@@ -1,32 +1,174 @@
...
+    marginHorizontal: 10,
+    backgroundColor: colors.white,
+    flex: 1,
+    borderColor: colors.lightGrey,
+    borderWidth: 1,
+    justifyContent: "flex-end"
+  },
+  section: {
+    borderBottomWidth: 1,
+    borderBottomColor: colors.grey,
```
Comment (zyadelgohary) [inline Dev/QConnect/src/screens/Evaluation/EvaluationPage.js]: Make sure to remove this line to remove the line as we discussed

File: Dev/QConnect/src/screens/Evaluation/EvaluationPage.js [hunk 4]
```diff
@@ -1,32 +1,174 @@
...
+                    ratingColor={colors.primaryLight}
+                    showRating={false}
+                  />
+                </View>
+              )} />
+
+            <TextInput
+              style={styles.notesStyle}
+              multiline={true}
+              numberOfLines={3}
```
Comment (zyadelgohary) [inline Dev/QConnect/src/screens/Evaluation/EvaluationPage.js]: Why are we limited the max number of lines. TextInput has the property of converting to a scrollable text box when the user types out of the box. (It becomes like a scrollView)

File: Dev/QConnect/src/screens/Evaluation/EvaluationPage.js [hunk 5]
```diff
@@ -1,32 +1,174 @@
...
+              placeholderColor={colors.black}
+            />
+          </View>
+
+        </View>
+
+        <View style={styles.buttonsContainer}>
+          <QcActionButton
+            text="Submit"
+            onPress={() => { }}
```
Comment (zyadelgohary) [inline Dev/QConnect/src/screens/Evaluation/EvaluationPage.js]: Add some kind of "to-do" comment reminding us to connect to redux state. I will take care of this with the final big PR of the app :)

File: Dev/QConnect/src/screens/Evaluation/EvaluationPage.js (hunk 4 above)
Comment (ElyDev) [reply Dev/QConnect/src/screens/Evaluation/EvaluationPage.js]: numberOfLines just makes the text scrollable after 3 lines. It is just another way to set height.

**Pull request closed without merging**